lib/jekyll-relative-links/generator.rb

```diff
@@ -6,7 +6,7 @@ class Generator < Jekyll::Generator
     include Jekyll::Filters::URLFilters
 
     LINK_TEXT_REGEX = %r!([^\]]+)!
-    FRAGMENT_REGEX = %r!(#.+)?!
+    FRAGMENT_REGEX = %r!(#(?:[^)]+|.+$))?!
```

Comment:
To be honest, easiest of all might be to kill the common FRAGMENT_REGEX snippet, and include different versions directly in each regex (inline vs reference).